packages/tailwindcss-language-server/src/language/cssServer.ts

```diff
@@ -354,6 +354,10 @@ function createVirtualCssDocument(textDocument: TextDocument): TextDocument {
     .replace(/@variants(\s+[^{]+){/g, replace())
     .replace(/@responsive(\s*){/g, replace())
     .replace(/@utility(\s+[^{]+){/g, replaceWithStyleRule())
+    .replace(/@custom-variant(\s+[^;]+);/g, (match: string) => {
+      let spaces = ' '.repeat(match.length - 11)
+      return `@media(p)${spaces}{}`
+    })
```

Comment:
```suggestion
    .replace(/@custom-variant(\s+[^;]+);/g, replace())
```

Why can't we use replace here? 🤔

Reply:
Because it produces `@media(p)${spaces}{`

I could add a flag to it to close the tag. I need to rework all of this at some point anyway.

Reply:
Oh yeah I'm stupid haha my bad.